### PR TITLE
test: add unit test for util.git.Client.CheckoutOrOrphan

### PR DESCRIPTION
**Pull request by daengdaengLee**
Hello. This is a PR for issue https://github.com/argoproj/argo-cd/issues/19071. (2nd / 5 methods)

This PR add unit tests for CheckoutOrOrphan method.
* (case 1) checkout to an existing branch
* (case 2) try to checkout to not existing branch, then create new orphan branch

If you notice anything I missed, please let me know in the comments, and I'll make the necessary corrections!

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: util/git/client_test.go [hunk 1]
```diff
@@ -334,3 +333,113 @@ func Test_nativeGitClient_SetAuthor(t *testing.T) {
...
+		err = runCmd(tempDir, "git", "checkout", baseBranch)
+		require.NoError(t, err)
+
+		// get base commit
+		gitCurrentCommitHash, err := outputCmd(tempDir, "git", "rev-parse", "HEAD")
+		require.NoError(t, err)
+		baseCommitHash := strings.TrimSpace(string(gitCurrentCommitHash))
+
+		out, err := client.CheckoutOrOrphan(expectedBranch, false)
+		require.NoError(t, err, "error output: ", out)
```
Comment (crenshaw-dev) [inline util/git/client_test.go:423]:
```suggestion
		// set the author for the initial commit of the orphan branch
		out, err := client.SetAuthor("test", "test@example.com")
		require.NoError(t, err, "error output: %s", out)

		out, err = client.CheckoutOrOrphan(expectedBranch, false)
		require.NoError(t, err, "error output: %s", out)
```

I had to do this to get it to pass locally.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks so much! One small suggestion.

File: util/git/client_test.go (hunk 1 above)
Comment (daengdaengLee) [reply util/git/client_test.go:425]: Thanks for your suggestion! I'll fix it.

Comment (daengdaengLee) [reply util/git/client_test.go]: I set author right after init git client. Could you possibly test it once more locally?

Comment (crenshaw-dev) [reply util/git/client_test.go]: Works now! I must have had an old commit.

**Review by crenshaw-dev — APPROVED** [review]
Looks great, thank you!

**Pull request merged**